pkg/rtc/mediatrack.go

```diff
@@ -652,3 +667,109 @@ func (t *MediaTrack) DebugInfo() map[string]interface{} {
 func (t *MediaTrack) Receiver() sfu.TrackReceiver {
 	return t.receiver
 }
+
+func (t *MediaTrack) OnSubscribedMaxQualityChange(f func(trackSid string, subscribedQualities []*livekit.SubscribedQuality) error) {
+	t.onSubscribedMaxQualityChange = f
+}
+
+func (t *MediaTrack) NotifySubscriberMute(subscriberID string) {
+	if !t.IsSimulcast() {
+		return
+	}
+
+	t.maxQualityLock.Lock()
+	_, ok := t.maxSubscriberQuality[subscriberID]
+	if !ok {
+		t.maxQualityLock.Unlock()
+		return
+	}
+
+	delete(t.maxSubscriberQuality, subscriberID)
+	t.maxQualityLock.Unlock()
+
+	t.updateQualityChange()
```

Comment:
Wanted to call this from here (and from the routine below) in a go routine, but had a couple of issues
1. Ordering
2. Tests. I tried writing tests with the use of goroutine and sync.WaitGroup, but just could not get it to work.

Reply:
Are you wanting to do so for purposes of deadlocks? The test challenges might be related to the need to lock within the goroutines spun up for testing?

Reply:
Yeah mostly for locking purposes. We are not locking here while doing the callback. So, I think we should be fine. I was also generally concerned about the callback having a long call chain and causing problems